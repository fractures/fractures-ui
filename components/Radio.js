import { Component } from "react"
import cc from "classcat"

const defaultProps = {
	isDisabled: false,
	label: "",
	set: () => null,
	state: null,
	value: null
}

class Radio extends Component {
	render() {
		const radioClasses = cc({
			"fr-radio flex flex-gap-1 relative": true,
			"fr-radio--disabled": this.props.isDisabled
		})

		return (
			<label className={ radioClasses }>
				<input
					type="radio"
					name="radio"
					className="h-0 w-0 none absolute"
					onChange={ () => this.props.set(this.props.value) }
					defaultChecked={ this.props.state === this.props.value }
				/>
				<span className="fr-radio__checkmark radius-max" />
				<span className="fr-radio__label">{this.props.label}</span>
				<style jsx global>{`
					.fr-radio {
						cursor: pointer;
						user-select: none;
					}

					.fr-radio__label {
						color: var(--shade-5);

						font-size: 0.875rem;
						font-weight: 500;
						line-height: 1.75rem;
					}

					.fr-radio__checkmark {
						height: 1.75rem;
						width: 1.75rem;

						background-color: var(--shade-0);
						border: 2px solid var(--shade-5);
					}

					.fr-radio input:checked ~ .fr-radio__checkmark {
						background-color: var(--shade-0);
					}

					.fr-radio__checkmark:after {
						position: absolute;
						top: 0.375rem;
						left: 0.375rem;

						opacity: 0;
						height: 1rem;
						width: 1rem;

						border-radius: 100%;
						background: var(--shade-5);
						transform: scale(0);
						transition: all 120ms ease-in-out;

						content: "";
					}

					.fr-radio input:checked ~ .fr-radio__checkmark:after {
						opacity: 1;

						transform: scale(1);
					}

					.fr-radio--disabled {
						opacity: 0.5;
						pointer-events: none;
					}
				`}</style>
			</label>
		)
	}
}

Radio.defaultProps = defaultProps

export default Radio
