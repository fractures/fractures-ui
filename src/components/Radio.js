import cc from "classcat"
import Label from "./shared/Label"
import React, { Component } from "react"

const defaultProps = {
	isDisabled: false,
	label: "",
	set: () => null,
	state: null,
	value: null
}

class Radio extends Component {
	render() {
		const radioClasses = cc({
			"fr-radio flex flex-gap-1 relative": true,
			"fr-radio--disabled": this.props.isDisabled
		})

		return (
			<label className={ radioClasses }>
				<input
					type="radio"
					name="radio"
					className="h-0 w-0 none absolute"
					onChange={ () => this.props.set(this.props.value) }
					defaultChecked={ this.props.state === this.props.value }
				/>
				<span className="fr-radio__checkmark radius-max" />
				<Label>{this.props.label}</Label>
				<style jsx global>{`
					.fr-radio {
						cursor: pointer;
						user-select: none;
					}

					.fr-radio__checkmark {
						height: 1.5rem;
						width: 1.5rem;

						background-color: var(--fr-ground);
						border: 2px solid var(--fr-500);
					}

					.fr-radio input:checked ~ .fr-radio__checkmark {
						background-color: var(--fr-ground);
					}

					.fr-radio__checkmark:after {
						position: absolute;
						top: 0.3125rem;
						left: 0.3125rem;

						opacity: 0;
						height: 0.875rem;
						width: 0.875rem;

						border-radius: 100%;
						background: var(--fr-500);
						transform: scale(0);
						transition: all 120ms ease-in-out;

						content: "";
					}

					.fr-radio input:checked ~ .fr-radio__checkmark:after {
						opacity: 1;

						transform: scale(1);
					}

					.fr-radio--disabled {
						opacity: 0.5;
						pointer-events: none;
					}
				`}</style>
			</label>
		)
	}
}

Radio.defaultProps = defaultProps

export default Radio
