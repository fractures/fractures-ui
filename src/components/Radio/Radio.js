import cc from "classcat"
import Label from "../Label/Label"
import PropTypes from "prop-types"
import React from "react"
import styled from "styled-components"

const propTypes = {
	className: PropTypes.string,
	isDisabled: PropTypes.bool,
	label: PropTypes.string,
	set: PropTypes.func.isRequired,
	state: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),
	value: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool])
}

const defaultProps = {
	isDisabled: false,
	label: "",
	set: () => null,
	state: false,
	value: ""
}

const NakedRadio = props => {
	const radioClasses = cc({
		"fr-radio--disabled": props.isDisabled
	})

	return (
		<label className={ `${ props.className } ${ radioClasses }` }>
			<input
				aria-checked={ props.state === props.value }
				type="radio"
				name="radio"
				onChange={ () => props.set(props.value) }
				defaultChecked={ props.state === props.value }
			/>
			<span className="fr-radio__checkmark" />
			<Label>{props.label}</Label>
		</label>
	)
}

const Radio = styled(NakedRadio)`
	position: relative;

	box-sizing: border-box;
	display: flex;

	cursor: pointer;
	user-select: none;

	&.fr-radio--disabled {
		opacity: 0.5;
		pointer-events: none;
	}

	input {
		display: none;
		height: 0;
		width: 0;

		visibility: hidden;
	}

	input:checked ~ .fr-radio__checkmark {
		background-color: var(--fr-ground);
	}

	.fr-radio__checkmark {
		box-sizing: inherit;
		height: 1.5rem;
		margin-right: 0.5rem;
		width: 1.5rem;

		background-color: var(--fr-ground);
		border: 2px solid var(--fr-500);
		border-radius: 100%;
	}

	.fr-radio__checkmark:after {
		position: absolute;
		top: 0.3125rem;
		left: 0.3125rem;

		opacity: 0;
		height: 0.875rem;
		width: 0.875rem;

		background: var(--fr-500);
		border-radius: 100%;
		transform: scale(0);
		transition: all 100ms ease-in-out;

		content: "";
	}

	input:checked ~ .fr-radio__checkmark:after {
		opacity: 1;

		transform: scale(1);
	}
`

NakedRadio.propTypes = propTypes
NakedRadio.defaultProps = defaultProps

Radio.propTypes = propTypes
Radio.defaultProps = defaultProps

export default Radio
