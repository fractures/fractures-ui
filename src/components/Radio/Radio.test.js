import { Radio } from "./Radio.js"

const mockFn = jest.fn()

describe("Radio", () => {
	const componentRender = <Radio set={ mockFn() } />
	const component = renderer.create(componentRender)
	const tree = component.toJSON()

	it("Renders correctly", () => {
		expect(tree).toMatchSnapshot()
	})
})
