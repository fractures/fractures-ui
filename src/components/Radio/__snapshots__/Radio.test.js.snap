// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Renders correctly 1`] = `
<label
  className="sc-bwzfXH dKZeNg "
>
  <input
    aria-checked={false}
    defaultChecked={false}
    name="radio"
    onChange={[Function]}
    type="radio"
  />
  <span
    className="fr-radio__checkmark"
  />
  <span
    className="sc-bdVaJa erEhky"
  >
    
  </span>
</label>
`;
